### ISSUE TITLE
config : `serviceaddress` tag doesn't recognize http server adddress

### ISSUE THREAD

**Issue by iwanbk** [zero-os/0-Disk#394]
```go
 invalid ZeroStorClusterConfig: Address: http://127.0.0.1:39555 does not validate as serviceaddress
```

**Comment by GlenDC** [comment]
Why do you give the `http://` in front, drop that and it will work.

**Comment by GlenDC** [comment]
I mean 0-stor doesn't work via a REST API does it? I thought it would be all just TCP/Unix?

**Comment by iwanbk** [comment]
It is REST & grpc

**Comment by GlenDC** [comment]
So the 0-stor cluster addresses will always be http(s) addresses?

**Comment by iwanbk** [comment]
REST use http

grpc is not and `serviceaddress` works fine with it

We are likely using grpc as it is faster

I was using REST to test something.

**Comment by GlenDC** [comment]
OK sure, but please try to communicate better when you're requesting features. It was never made clear that the address should be able to be a URL, TCP Address and Unix Socket. If that's what you want, than it can be added for sure.

**Comment by iwanbk** [comment]
> OK sure, but please try to communicate better when you're requesting features. It was never made clear that the address should be able to be a URL, TCP Address and Unix Socket

Please note that i only asked the validation to be non empty string 
https://github.com/zero-os/0-Disk/issues/360#issuecomment-319928518

**Comment by GlenDC** [comment]
That's what you asked sure, but I don't agree with such low validation as it is not helpful at all in case you make mistakes. The point of validation is that we try to catch as much possible errors as we can upfront, before we do anything else. This is especially important once other people start using it (such as Egypt), who really require strict validation, as they'll otherwise ask 999 questions about why doesn't this work, while with validation in place we avoid many of those questions.

And that's by the way referring to a comment of yours on the original issue. When you reported the previous issue (#387), you definitely had another change to explain what type of addresses it could contain. But instead you choose to just report the unix issue, to than also request http the next day. If you had just requested that at the same time, you wouldn't have gotten that issue today. You know, communication.

**Comment by iwanbk** [comment]
> Non empty value for all fields is fine for now

please note `for now` here, as i really didn't sure which validation which should be supported.

For example, the unix socket address was just added 4 days ago 
https://github.com/zero-os/0-stor/pull/148/commits/cb52d2b0f29ef97e00edbdb89f194308aafd00c7


> When you reported the previous issue (#387), you definitely had another change to explain what type of addresses it could contain. But instead you choose to just report the unix issue, to than also request http the next day. If you had just requested that at the same time, you wouldn't have gotten that issue today. You know, communication


Well, at the time,  i had no idea that your validation won't support `http`.

**Comment by GlenDC** [comment]
Well now you know for the future. Validation is as strict as possible. So if you don't mention it needs to support http addresses, it won't be accepted. Now you do tell me, so it will be accepted.

**Comment by iwanbk** [comment]
> Well now you know for the future. Validation is as strict as possible

Yeah i know, but once again, i didn't have spec at that time, so i only asked for non empty string.

Other than `unix` address that was just added 4 days ago, grpc client itself just added 5 days ago
https://github.com/zero-os/0-stor/pull/141

So, please don't get mad for such feature/change request, for something you build based on your own assumption.

**Comment by iwanbk** [comment]
labeled it as `type_feature` instead of `type_bug` as it is not really a bug

**Comment by iwanbk** [comment]
> And this new type should only be used for the 0-Stor server addresses. Not for the metadata stuff, as that's etcd.

Missed this part of comment. Http address also valid for etcd

**Comment by chrisvdg** [comment]
> @chrisvdg: this [url support] shouldn't be added to the service Address though. You can make a new type for this, which encapsulates the Service Address and which also accepts a URL (using IsURL). Not sure what a good name for it is. And this new type should only be used for the 0-Stor server addresses. Not for the metadata stuff, as that's etcd.

>Missed this part of comment. Http address also valid for etcd

@GlenDC, so should it still be added to a new type and validation or can it be added to the current serviceaddress validation?

**Comment by GlenDC** [comment]
@iwanbk please close this issue as soon as you validated this issue is solved using PR #396.

**Comment by iwanbk** [comment]
> @iwanbk please close this issue as soon as you validated this issue is solved using PR #396.

Closed, was off yesterday for independence day celebration, hence the delay
